firmware/src/controller/controller.c

```diff
@@ -137,6 +140,22 @@ PAC5XXX_RAMFUNC void CLControlStep(void)
         }
     }
 
+    const float vel_setpoint_delta = state.vel_setpoint_goal - state.vel_setpoint;
+    if (abs(vel_setpoint_delta) <= config.vel_increment)
+    {
+        state.vel_setpoint = state.vel_setpoint_goal;
+    }
+    else
+    {
+        if (vel_setpoint_delta < 0)
+        {
+            state.vel_setpoint -= config.vel_increment;
+        }
+        else 
+        {
+            state.vel_setpoint += config.vel_increment;
+        }
+    }
```

Comment:
I'm wondering if this could be replaced by something like below:
`state.vel_setpoint += our_clamp(state.vel_setpoint_goal - state.vel_setpoint, -config.vel_increment, config.vel_increment);`

the function call will be inlined, and there are only two conditions. What do you think?